docs/developer/branching-and-workflow.md

```diff
@@ -0,0 +1,31 @@
+# Branching and Workflow
+
+NKG is an open source and public repository; our goal is for a spartan and easily manageable set of branches. Ideally,
+our branches are created minimally; there will be a main branch and a handful of release and other long-term feature
+branches.
+
+Internal developers and external contributors will follow a fork and merge process. Each contributor should fork the
+repo to their own space; branch, experiment, develop and merge from their space into NKG’s main branch. NKG branches are
+long-lived, developer branches are ephemeral and unconstrained outside the main repository.
```

Comment:
Suggestion:

Internal developers and external contributors will follow a fork and merge process. Each contributor should fork the
repo to their own space; branch, experiment, develop and prepare a PR to merge their work into NKG’s main branch. This way ephemeral developer branches will remain outside of the main repository.